spec.emu

```diff
@@ -390,6 +465,29 @@ contributors: Google, Ecma International
     </emu-note>
   </emu-clause>
 
+  <emu-clause id="sup-temporal-calendardatearithmeticyear" type="abstract operation">
+    <h1>
+      CalendarDateArithmeticYear (
+        _calendar_: a String,
+        _date_: a Temporal.PlainDateTime, Temporal.PlainDate, or Temporal.PlainYearMonth,
+      ): an integer
+    </h1>
+    <dl class="header">
+      <dt>description</dt>
+      <dd>It performs implementation-defined processing to find the year for the date corresponding to _date_ in the context of the calendar represented by _calendar_ relative to a well-defined epoch year for that calendar.</dd>
+    </dl>
+    <emu-alg>
+      1. Assert: CalendarSupportsEra(_calendar_) is *true*.
+      1. Let _year_ be _date_.[[ISOYear]].
+      1. Let _r_ be the row in <emu-xref href="#table-epoch-years"></emu-xref> which the value of the Calendar column is _calendar_.
+      1. Let _epochYear_ be the value given in the *"Epoch Year"* column of _r_.
+      1. Let _relativeYear_ be _year_ - _epochYear_.
+      1. Let _arithmeticYear_ be the year that according to implementation-defined processing is the year of the calendar represented by _calendar_ that started during the ISO year _relativeYear_.
```

Comment:
There's something off about this logic. For one thing I don't think _relativeYear_ is an ISO year after the epoch year is subtracted from it. But also, the arithmetic year may not even be numbered as a calendar year (e.g. in the Japanese calendar) so we cannot say "the year of the calendar that started during ..."

I think it has to be something like "Let _arithmeticYear_ be the number of whole years in the calendar represented by _calendar_ elapsed since the calendar year that started during ISO year _epochYear_, according to implementation-defined processing."

(Not sure if that language would correctly cover negative arithmetic years.)

Reply:
Yeah, the calendar's ISO Year should not be subtracted against the Epoch Year, because the difference is always in terms of solar years.